src/screens/Customers/CustomerMain.tsx

```diff
@@ -78,39 +88,53 @@ function CustomerMain(props: RouteComponentProps & UserProps) {
   const getCustomers = () => {
     const siteData = store.getState().siteData.currentSite;
     const allCustomers: CustomerRecord[] = siteData.customers;
-    allCustomers.sort(sortByFunction());
-    allCustomers.filter(filterByFunction);
-    setFilteredCustomers(allCustomers);
+    allCustomers.sort(sortByFunction);
+    sortAndFilterCustomers(allCustomers)
     setFullCustomers(allCustomers);
     const customersTrie: TrieTree<CustomerRecord> = new TrieTree('name');
     customersTrie.addAll(allCustomers);
     setAllCustomersTrie(customersTrie);
   }
 
-  const sortByFunction = (): (a: CustomerRecord, b: CustomerRecord) => number => {
+  const sortAndFilterCustomers = (customers: CustomerRecord[]) => {
+    const groupA = customers.filter(filterByFunction);
+    const groupB = customers.filter((customer: CustomerRecord) => !filterByFunction(customer));
+    setFilteredCustomers(groupA);
+    setFilteredCustomersAlt(groupB); 
+  }
+
+  /**
+   * Comparator function for sorting two customer records. 
+   * It is conditioned by the current user selection for sorting. 
+   */
+  const sortByFunction = (customerA: CustomerRecord, customerB: CustomerRecord): number => {
     switch (sortBy) {
-      case (SortBy.meter): {
-        return (a: CustomerRecord, b: CustomerRecord) => { return a.meterNumber - b.meterNumber }
+      case (SortBy.METER): {
+        return customerA.meterNumber - customerB.meterNumber;
       }
-      case (SortBy.name): {
-        return (a: CustomerRecord, b: CustomerRecord) => { return a.name.localeCompare(b.name) }
+      case (SortBy.NAME): {
+        return customerA.name.localeCompare(customerB.name)
       }
       default: {
-        return (a: CustomerRecord, b: CustomerRecord) => { return 0 }
+        return 0
       }
     }
   }
 
-  const filterByFunction = (e: CustomerRecord): boolean => {
+  /**
+   * Comparator function for filtering customer records. 
+   * It is conditioned by the current user selection for filtering. 
+   */
+  const filterByFunction = (customer: CustomerRecord): boolean => {
     switch (filterBy) {
-      case (FilterBy.meterStatus): {
-        return e.hasmeter
+      case (FilterBy.METER_STATUS): {
+        return customer.hasmeter
       }
-      case (FilterBy.paymentStatus): {
-        return e.payments.length > 0
+      case (FilterBy.PAYMENT_STATUS): {
+        return customer.payments.length > 0
```

Comment:
I should clarify with Ashley exactly how this should be sorted (whether it's if people have paid this period or if it's related to outstanding balance).

Can we change this so that it filters customers that have a zero outstanding balance & people with a non-zero outstanding balance? The payments array is related to all the payments the customer has made in their lifetime as a customer.

Reply:
Gotcha! After meeting w Ashley, your intuition is right that it's regarding outstanding balance. My next commit will include this